src/workloads/query/TimeseriesTSBSQuery.yml

```diff
@@ -0,0 +1,1188 @@
+          Operations:
+            - OperationName: aggregate
+              OperationCommand:
+                Pipeline:
+                  # All usage_* fields are random integers between 0-100.
+                  [
+                    {
+                      $match:
+                        {
+                          $and:
+                            [
+                              { usage_steal: { $gte: 25, $lt: 85 } },
+                              { "tags.hostname": { $in: *HostName8 } },
+                              {
+                                time:
+                                  {
+                                    $gte:
+                                      {
+                                        ^Cycle:
+                                          {
+                                            ofLength: 10,
+                                            fromGenerator:
+                                              {
+                                                ^IncDate:
+                                                  {
+                                                    start: *StartDate,
+                                                    step: 2520000,
+                                                  },
+                                              },
+                                          },
+                                      },
+                                    $lte:
+                                      {
+                                        ^Cycle:
+                                          {
+                                            ofLength: 10,
+                                            fromGenerator:
+                                              {
+                                                ^IncDate:
+                                                  {
+                                                    start: *Hour12,
+                                                    step: 2520000,
+                                                  },
+                                              },
+                                          },
+                                      },
+                                  },
+                              },
+                            ],
+                        },
+                    },
+                    {
+                      $group:
+                        {
+                          _id: { $dateTrunc: { date: "$time", unit: "hour" } },
+                          min_usage_softirq: { $min: "$usage_softirq" },
+                          min__usage_steal: { $min: "$usage_steal" },
+                          min_usage_guest: { $min: "$usage_guest" },
+                          min_usage_user: { $min: "$usage_user" },
+                          min_usage_system: { $min: "$usage_system" },
+                        },
+                    },
+                    { $sort: { _id: 1 } },
+                  ]
```

Comment:
Other optimizations that we should cover (not necessarily on this dataset, but if we can do it here, so much the better):
1. whole bucket filter
2. empty include list on the unpack stage
3. count/min/max rewrites (unless there are already workloads for these)

Reply:
let's move these into the "optimizations" workload as well (after all, bucket-targeting is a form of optimization :))
btw, what's the workload for `$count`?